include/domain-types.h

```diff
@@ -16,14 +17,74 @@ namespace connect
 
 typedef int32_t id_t;
 
-typedef int status_t;
-
-enum
+class Status
 {
-    STATUS_OK       = 0,
-    STATUS_ERROR    = 1
+public:
+    Status(int code, bool isError, int facility)
+    {
+        assert(code - (code & CODE_MASK) == 0);
+        assert(facility - (facility & FACILITY_MASK) == 0);
+
+        status_ = ((uint32_t)(isError ? 1 : 0) << 31) | ((uint32_t)facility << 16) | (uint32_t)code;
+    }
+
+    // codes
+    enum
+    {
+        SUCCESS = 0,
+        FAILURE = 1 // for any (unknown) reason
+
+        // extend with more specific codes as necessary
+    };
+
+    enum
+    {
+        FACILITY_NONE = 0, // for general codes
+        FACILITY_NETWORK = 1,
+        FACILITY_WEBAPI = 2
```

Comment:
No.
Status may have only one facility associated with it. Think of it as status origin. Origin can be only one.